Building Zig + C projects fail on macOS

The main issue seems to be a regression in Zig's ability to build and link c/c++ code. A few months back there was a change made to zig that I haven't yet pin-pointed. Since that time Zig has been unable to build an executable in a project with C++ code in it (C still worked at that time). Using today's master, neither C nor C++ code in a Zig project will build.

Some Example projects with Zig and C/C++ code that work with previous versions of Zig (`0.6.0+68bf29c31` is the last working version I have available):
https://github.com/floooh/sokol-zig
https://github.com/prime31/zig-upaya
https://github.com/prime31/zig-miniaudio

The errors are generally the same for all projects. Here is one example:
```
error(compilation): clang failed with stderr: In file included from zig-master/build/lib/zig/libcxx/src/algorithm.cpp:9:
In file included from zig-master/build/lib/zig/libcxx/include/algorithm:643:
In file included from zig-master/build/lib/zig/libcxx/include/memory:681:
In file included from zig-master/build/lib/zig/libcxx/include/atomic:571:
In file included from zig-master/build/lib/zig/libcxx/include/__threading_support:27:
In file included from zig-master/build/lib/zig/libc/include/generic-glibc/pthread.h:22:
In file included from zig-master/build/lib/zig/libc/include/generic-glibc/sched.h:25:
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:139:3: error: 
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:145:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:145:24: error: expected ';' after top level declarator
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:146:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:147:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:148:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:149:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:149:26: error: expected ';' after top level declarator
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:150:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:151:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:151:26: error: expected ';' after top level declarator
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:152:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:153:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:153:26: error: expected ';' after top level declarator
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:154:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:155:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:156:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:157:1: error: unknown type name '__STD_TYPE'
zig-master/build/lib/zig/libc/include/generic-glibc/bits/types.h:158:1: error: unknown type name '__STD_TYPE'
fatal error: too many errors emitted, stopping now [-ferror-limit=]


... many of these for 40 or so files in zig/libcxx/src

error(compilation): zig-master/build/lib/zig/libcxx/src/algorithm.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/any.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/atomic.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/barrier.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/bind.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/chrono.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/condition_variable.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/condition_variable_destructor.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/debug.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/experimental/memory_resource.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/filesystem/directory_iterator.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/filesystem/operations.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/functional.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/future.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/hash.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/ios.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/iostream.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/locale.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/memory.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/mutex.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/mutex_destructor.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/new.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/optional.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/random.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/random_shuffle.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/regex.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/shared_mutex.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/stdexcept.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/string.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/strstream.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/system_error.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/thread.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/valarray.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/variant.cpp:1:1: unable to build C object: clang exited with code 1
error(compilation): zig-master/build/lib/zig/libcxx/src/vector.cpp:1:1: unable to build C object: clang exited with code 1
error: unable to build libcxx: BuildingLibCObjectFailed
```

--- a/src/Compilation.zig
+++ b/src/Compilation.zig
@@ -2127,7 +2127,9 @@ fn detectLibCIncludeDirs(
         return detectLibCFromLibCInstallation(arena, target, lci);
     }
 
-    if (target_util.canBuildLibC(target)) {
+    if (target_util.canBuildLibC(target)) outer: {
+        if (is_native_os and target.isDarwin()) break :outer; // If we're on Darwin, we want to use native since we only have headers.
+
         const generic_name = target_util.libCGenericName(target);
         // Some architectures are handled by the same set of headers.
         const arch_name = if (target.abi.isMusl()) target_util.archMuslName(target.cpu.arch) else @tagName(target.cpu.arch);

